ChatSecure/Classes/Model/Yap Storage/OTRXMPPRoomOccupant.swift

```diff
@@ -161,8 +176,9 @@ public extension OTRXMPPRoomOccupant {
         
         var parameters: [String] = [roomUniqueId]
         var queryString = "Where \(RoomOccupantIndexColumnName.roomUniqueId) == ? AND ("
-        parameters.append(jid.full)
-        queryString.append("\(RoomOccupantIndexColumnName.jid) == ?")
+        // We build the secondary index with appended \0 to avoid matching wrong jids, so add a matching \0s here.
+        parameters.append("%\t\(jid.full)\t%")
+        queryString.append("\(RoomOccupantIndexColumnName.jids) LIKE ?")
```

Comment:
Although it's probably not ideal, if it works as intended I think we should move forward for now until we think of a better/faster solution.